tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -78,6 +52,7 @@ inline void debug_sanitize_post_noc_addr_and_hang(uint64_t a, uint32_t l, uint32
         v[noc_index].len = l;
         v[noc_index].which = debug_get_which_riscv();
         v[noc_index].invalid = invalid;
+        v[noc_index].invalid_reason = reason;
```

Comment:
ok, sorry for not thinking of this earlier...I think the reason should be decided on the host.  reason is: this will bloat code on the device.

Reply:
Part of #6738 changes, removed from this PR